lib/projectParser.ts

```diff
@@ -274,8 +333,54 @@ export class ProjectParser {
     this.cachedElaborate = filter;
 
   }
+
+  public findSettings(url: URL): FileCacheSettings | undefined {
+    let path = fileURLToPath(url);
+    while (path.length > 1) {
```

Comment:
this is to avoid infinite loop when reaching / ?
Does this work on windows? Is there a os agnostic version of this?